calicoctl/commands/node/diags.go

```diff
@@ -235,3 +243,39 @@ func writeDiags(cmds diagCmd, dir string) {
 		log.Errorf("Error writing diags to file: %s\n", err)
 	}
 }
+
+func walkIPTables(file string, command string)([]byte) {
+
+	tablesFile, err := os.Open(file)
+	if err != nil {
+		fmt.Printf("Failed to open: %s\n", file)
+	}
+
+	command, err = exec.LookPath(command)
+	if err != nil {
+		fmt.Printf("Failed to run command: %s\nError: Not found in PATH\n", command)
+	}
+
+	scanner := bufio.NewScanner(tablesFile)
+	scanner.Split(bufio.ScanLines)
+
+	var persist string
+	for scanner.Scan() {
+		table := scanner.Text()
+		cmd := exec.Command(command, "-S", "-t", table)
+
+		var rules, errbuf bytes.Buffer
+		cmd.Stdout = &rules
+		cmd.Stderr = &errbuf
+		err = cmd.Run()
+
+		if err != nil {
+			fmt.Printf("ERROR in %s: %s", table, errbuf.String())
+		} else {
+			persist += fmt.Sprintf("*%s\n%sCOMMIT\n", table, rules.String())
+		}
+	}
+
+	tablesFile.Close()
```

Comment:
It's more common in go to use `defer` for closing files, especially when you might have several exit paths to a function.

You can just do `defer tablesFile.Close()` right after opening the file to make sure it will always get closed.